ddtrace/tracer/textmap.go

```diff
@@ -114,74 +115,188 @@ func NewPropagator(cfg *PropagatorConfig) Propagator {
 	if cfg.PriorityHeader == "" {
 		cfg.PriorityHeader = DefaultPriorityHeader
 	}
-	return &propagator{cfg}
+	return &propagator{
+		injectors:  makeInjectors(cfg),
+		extractors: makeExtractors(cfg),
+	}
 }
 
-// propagator implements a propagator which uses TextMap internally.
+// propagator implements a Propagator that supports TextMap carriers.
 // It propagates the trace and span IDs, as well as the baggage from the
 // context.
-type propagator struct{ cfg *PropagatorConfig }
+type propagator struct {
+	injectors  []textMapPropagator
+	extractors []textMapPropagator
+}
+
+// makeInjectors returns a list of injectors to apply when propagating a
+// context. By default, only the datadog propagation style will be used.
```

Comment:
I'm not so convinced that the word "style" quite expresses what we want here. I'd like to suggest changing this term throughout to one of the following (my preference order goes first to last):
* propagation format
* propagation standard
* propagation protocol

"Style" doesn't quite cut it and can be misleading imho.

Reply:
It's the term used in the existing implementations.

Reply:
We don’t necessarily have to do the same. Different languages have different idioms. I still feel that all the other options are more appropriate than style. I’d use “format” if it was me. If you feel strongly about it, leave it, but I think “style” is a poor choice.